### DISCUSSION TITLE
lockedVersion is null for packagefileVersion=3

### DISCUSSION THREAD

**Discussion by mjunker** [renovatebot/renovate#21550] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.46.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It never worked

### Wanted end result.

When using a package-lock with lockfileVersion 3 Renovate does not create merge request. It should work the same way as it does with lockfileVersion 2.

### What you tried so far.

I compared the debug logs when using lockfileVersion 2 and 3 and the only difference seems to be it cannot read the lockedVersion of the dependency because in the debug logs i see that "lockedVersion" seems to be null (see attached log). When using lockfileVersion 2 this is set to the version in package-lock.

I created a repo where you can reproduce the bug: https://github.com/mjunker/renovate-lockVersion3

It contains angular/core which is locked at 14.0.1. based on the package rules I would expect it to be renovated to version 14.3.0. When I regenerate the package-lock with npm 7 and packagelockVersion 2 then the pull request is created as expected and I can see the the lockedVersion was resolved.

### Relevant debug logs

<details><summary>Logs</summary>

```
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "rangeStrategy": "in-range-only",
      "enabled": false
    },
    {
      "matchPackageNames": [
        "@angular/core"
      ],
      "rangeStrategy": "in-range-only",
      "enabled": true,
      "allowedVersions": "<=14.3.0"
    },
    {
      "matchPackagePatterns": [
        "^@angular/"
      ],
      "rangeStrategy": "in-range-only",
      "enabled": true,
      "groupName": "@angular libraries"
    }
  ]
}
DEBUG: packageFiles with updates (repository=mjunker/renovate-lockVersion3, baseBranch=main)
       "config": {
         "npm": [
           {
             "deps": [
               {
                 "depType": "dependencies",
                 "depName": "@angular/core",
                 "currentValue": "^14.0.1",
                 "datasource": "npm",
                 "prettyDepType": "dependency",
                 "lockedVersion": null,
                 "updates": [],
                 "packageName": "@angular/core",
                 "warnings": [],
                 "versioning": "npm",
                 "sourceUrl": "https://github.com/angular/angular",
                 "registryUrl": "https://registry.npmjs.org",
                 "sourceDirectory": "packages/core",
                 "currentVersion": "14.3.0"
               }
             ],
             "packageFileVersion": "1.0.0",
             "npmrc": "***********",
             "managerData": {
               "npmLock": "project/package-lock.json",
               "packageJsonName": "renovate-test-project",
               "yarnZeroInstall": false,
               "hasPackageManager": false
             },
             "skipInstalls": true,
             "packageFile": "project/package.json",
             "lockFiles": ["project/package-lock.json"],
             "constraints": {}
           }
         ]
       }```

</details>

**Comment by HonkingGoose** [comment]
> ### If you're self-hosting Renovate, tell us what version of Renovate you run.
> 
> 12.30.0

I hope that's just a typo, that version is from 2018-05-19. If you're _really_ using that version of Renovate, please update Renovate first!

**Comment by HonkingGoose** [comment]
I've forked the reproduction into our reproduction organization: https://github.com/renovate-reproductions/21550

**Comment by mjunker** [comment]
Ah yes I confused the version numbers. Updated the one we use:

 INFO: Repository started (repository=mjunker/renovate-lockVersion3)
       "renovateVersion": "35.46.0"

**Comment by rarkins** [comment, marked as answer]
Hi, it looks like Renovate does not yet support the lockfileVersion=3 format for npm. Please file this as a _feature request_ issue, linking to your reproduction. Suggested issue title: "npm: support parsing lockfileVersion=3"

**Comment by mjunker** [comment]
Created the feature request: https://github.com/renovatebot/renovate/issues/21552
